src/upload-helper.ts

```diff
@@ -47,12 +47,13 @@ export class UploadHelper {
     bucketName: string,
     filename: string,
     destination?: string,
+    gzip?: boolean,
   ): Promise<UploadResponse> {
     interface UploadOptions {
       gzip: boolean;
       destination?: string;
     }
-    const options: UploadOptions = { gzip: true };
+    const options: UploadOptions = { gzip: !!gzip };
```

Comment:
Shouldn't this just be `{ gzip: gzip }` or even `{ gzip }`?

Reply:
Since gzip is required above I get the following error: `Type 'boolean | undefined' is not assignable to type 'boolean'` 
I changed it to `gzip: gzip || true` which reads better

Reply:
Hmm, since gzip is required above, I needed to set a default. I went with the more readable `gzip: gzip === undefined ? true : gzip`